File: src/test/java/com/runninglane/dto/buddy/test/java/cases/inner/InnerDtoInterfaceTest.java
package com.runninglane.dto.buddy.test.java.cases.inner;

import com.runninglane.dto.buddy.DtoBuddy;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class InnerDtoInterfaceTest {

    private final DtoBuddy dtoBuddy = new DtoBuddy();
    
    @Test
    public void testImplement()
    {
        Class<?> concreteClass = dtoBuddy.implementor()
            .withBaseClass(InnerDtoInterfaceEncloser.InnerDtoInterface.class)
            .implement();
        assertNotNull(concreteClass);
        assertTrue(InnerDtoInterfaceEncloser.InnerDtoInterface.class. isAssignableFrom (concreteClass));
    }

}

File: src/test/kotlin/com/runninglane/dto/buddy/test/cases/inner/InnerDtoInterfaceTest.kt
package com.runninglane.dto.buddy.test.cases.inner

import com.runninglane.dto.buddy.DtoBuddy
import kotlin.reflect.full.isSubclassOf
import kotlin.test.Test
import kotlin.test.assertNotNull
import kotlin.test.assertTrue

class InnerDtoInterfaceTest {
    private val dtoBuddy = DtoBuddy()

    @Test
    fun testImplement() {
        val concreteClass = dtoBuddy.implement(InnerDtoInterfaceEncloser.InnerDtoInterface::class)
        assertNotNull(concreteClass)
        assertTrue(concreteClass.isSubclassOf(InnerDtoInterfaceEncloser.InnerDtoInterface::class))
    }

}